frontend/src/metabase/admin/settings/components/widgets/SiteUrlWidget.tsx

```diff
@@ -0,0 +1,62 @@
+import type { ChangeEvent } from "react";
+import { t } from "ttag";
+
+import { useAdminSetting } from "metabase/api";
+import { useHasTokenFeature, useToast } from "metabase/common/hooks";
+import InputWithSelectPrefix from "metabase/components/InputWithSelectPrefix";
+import type { GenericErrorResponse } from "metabase/lib/errors";
+import { Box } from "metabase/ui";
+
+import { SettingHeader } from "../SettingHeader";
+
+export function SiteUrlWidget() {
+  const { value, updateSetting } = useAdminSetting("site-url");
+  const isHosted = useHasTokenFeature("hosting");
```

Comment:
I was surprised to see this method of getting `isHosted`. We seem to have lots of ways to get settings, 
1. const isHosted = useSetting("is-hosted?");
2. const isHosted = useSelector(getIsHosted);
3. const isHosted = useHasTokenFeature("hosting");

[I asked the engineering-fe chat about it](https://metaboat.slack.com/archives/C505ZNNH4/p1742954660394039)